### PR TITLE
Lint.job runner specific timeout

### LINKED ISSUE #6131: Lint: Rule-Request: Avoid Job Runner Wallclock Directives
Users should be encouraged to use `execution time limit` in preference to `[directives]--this-runners-time-limit` so that Cylc can be aware of how long a job should take, and handle comms failures gracefully.

### PR DESCRIPTION
**Pull request by wxtim**
Closes #6131 

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: tests/unit/scripts/test_lint.py [hunk 1]
```diff
@@ -178,6 +178,8 @@
 something\t
     [[bar]]
         platform = $(some-script foo)
+            [[[dependencies]]]
+                -l walltime 666
```
Comment (MetRonnie) [inline tests/unit/scripts/test_lint.py:182]:
```suggestion
            [[[directives]]]
                -l = walltime 666
```

**Review by MetRonnie — COMMENTED** [review]
I don't get any errors for this
```
[runtime]
    [[foo]]
        [[[directives]]]
            -l = walltime 666
```

**Comment by MetRonnie** [comment]
Marked my review comment as resolved but then it's hidden all the comments on the files, which are **not** resolved. Now I can't change the review comment to unresolved 😑

**Comment by MetRonnie** [comment]
Also should remove any instances of time limit directives from the docs? E.g. https://github.com/cylc/cylc-flow/blob/86d24c3c1e63806590a89e75cc4202851feef1af/cylc/flow/job_runner_handlers/sge.py#L40

File: cylc/flow/scripts/lint.py [hunk 2]
```diff
@@ -533,6 +564,30 @@ def list_wrapper(line: str, check: Callable) -> Optional[Dict[str, str]]:
...
+        'short': (
+            'Use ``[runtime][TASK]execution time limit``'
+            ' rather than job runner directive: ``{directive}``.'
+        ),
+        'rst': (
+            'Using ``[runtime][TASK]execution time limit`` is'
+            ' recommended in preference to using job runner'
+            ' directives because it allows Cylc to retain awareness'
+            ' of whether the job should have finished, even if contact'
+            ' with the target job runner\'s platform has been lost.'
```
[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/flow/scripts/lint.py:578]: It's  a reasonable point... mind if I leave it?

**Comment by wxtim** [comment]
> Also should remove any instances of time limit directives from the docs? E.g.
> 
> https://github.com/cylc/cylc-flow/blob/86d24c3c1e63806590a89e75cc4202851feef1af/cylc/flow/job_runner_handlers/sge.py#L40

This appears to only be an issue for SGE where a not below tells you not to do precisely this!

n.b. Some of these directives still appear in documentation where examples of the job file show how the `execution time limit` will be converted to directives.

**Review by MetRonnie — APPROVED** [review]
The only thing it doesn't pick up on is when multiple directives are specified on the same setting, e.g.
```
[[[directives]]]
    -l = walltime=60:nodes=1
```
But otherwise I have tested and am happy with this

**Comment by wxtim** [comment]
> The only thing it doesn't pick up on is when multiple directives are specified on the same setting, e.g.
> 
> ```
> [[[directives]]]
>     -l = walltime=60:nodes=1
> ```
> 
> But otherwise I have tested and am happy with this

Not a pattern we see too often

File: cylc/flow/scripts/lint.py [hunk 3]
```diff
@@ -162,6 +165,34 @@
...
+            None
+        )
+        if directive:
+            directives[module.name] = directive
+    return directives
+
+
+WALLCLOCK_DIRECTIVES = get_wallclock_directives()
+
+
```
Comment (markgrahamdawson) [inline cylc/flow/scripts/lint.py:185]: Should this be check_wallclock_directives() now?

Comment (MetRonnie) [reply cylc/flow/scripts/lint.py:185]: Are you referring to next line? https://github.com/cylc/cylc-flow/blob/b42ff8b3c7bfc148a2c5d81d4e01833a431b276f/cylc/flow/scripts/lint.py#L186

Comment (markgrahamdawson) [reply cylc/flow/scripts/lint.py:185]: yea

Comment (wxtim) [reply cylc/flow/scripts/lint.py:185]: Legit point

File: cylc/flow/scripts/lint.py [hunk 4]
```diff
@@ -533,6 +564,30 @@ def list_wrapper(line: str, check: Callable) -> Optional[Dict[str, str]]:
...
+        )
+        + '\n * '.join((
+            f'``{directive}`` ({job_runner})'
+            for job_runner, directive in WALLCLOCK_DIRECTIVES.items()
+        )) + (
+            '\n\n.. note:: Using ``execution time limit`` which'
+            ' is automatically translated to the job runner\'s timeout'
+            ' directive can make your workflow more portable.'
+        ),
+        FUNCTION: check_etl_directives,
```
Comment (wxtim) [inline cylc/flow/scripts/lint.py:590]:
```suggestion
        FUNCTION: check_wallclock_directives,
```

File: cylc/flow/scripts/lint.py [hunk 5]
```diff
@@ -162,6 +165,34 @@
...
+        )
+        if directive:
+            directives[module.name] = directive
+    return directives
+
+
+WALLCLOCK_DIRECTIVES = get_wallclock_directives()
+
+
+def check_etl_directives(line: str) -> Union[Dict[str, str], bool]:
```
Comment (wxtim) [inline cylc/flow/scripts/lint.py:186]:
```suggestion
def check_wallclock_directives(line: str) -> Union[Dict[str, str], bool]:
```

File: cylc/flow/scripts/lint.py (hunk 3 above)
Comment (wxtim) [reply cylc/flow/scripts/lint.py:185]: Have dealt with.

**Review by markgrahamdawson — APPROVED** [review]

**Pull request merged**